tests/e2e/cmp_test.go

```diff
@@ -54,6 +54,11 @@ var _ = Describe("odoCmpE2e", func() {
 			runCmd("odo app create " + appTestName)
 		})
 
+		It("should show an error when ref flag is provided with sources except git", func() {
```

Comment:
I would like to see a positive case as well :)
Negatives can be tested with in UT(in the interest of keeping the number of tests minimum) ;)